src/mattack_actors.cpp

```diff
@@ -573,3 +577,91 @@ void gun_actor::shoot( monster &z, Creature &target, const gun_mode_id &mode ) c
         target.add_effect( effect_was_laserlocked, 5_turns );
     }
 }
+
+void throw_actor::load_internal( const JsonObject &obj, const std::string & )
+{
+    if( obj.has_array( "damage_max_instance" ) ) {
+        damage_max_instance = load_damage_instance( obj.get_array( "damage_max_instance" ) );
+    } else if( obj.has_object( "damage_max_instance" ) ) {
+        damage_max_instance = load_damage_instance( obj );
+    }
+
+    range = obj.get_float( "range", 5.0f );
+    speed = obj.get_int( "speed", 30 );
+    move_cost = obj.get_int( "move_cost", 150 );
+
+    obj.read( "description", description );
+}
+
+std::unique_ptr<mattack_actor> throw_actor::clone() const
+{
+    return std::make_unique<throw_actor>( *this );
+}
+
+bool throw_actor::call( monster &z ) const
+{
+    if( !z.can_act() || !z.move_effects( false ) ) {
+        return false;
+    }
+
+    Creature *target = z.attack_target();
+    if( !target ) {
+        return false;
+    }
+
+    if( rl_dist( z.pos(), target->pos() ) > range || !z.sees( *target ) ) {
+        return false;
+    }
+
+    const itype_id throwable_type = z.type->starting_ammo.begin()->first;
+
+    if( z.ammo[throwable_type] == 0 ) {
+        return false;
+    }
+
+    z.moves -= move_cost;
+    add_msg_if_player_sees( z, m_bad, description.translated() );
+    z.ammo[throwable_type] -= 1;
+
+    item to_throw( throwable_type, calendar::turn, 1 );
+
+    if( mattack::dodge_check( &z, target ) || target->uncanny_dodge() ) {
+        game_message_type msg_type = target->is_avatar() ? m_good : m_info;
+        target->add_msg_player_or_npc( msg_type,
+                                       _( "You dodge the thrown %1$s!" ),
+                                       _( "<npcname> dodges the thrown %1$s!" ),
+                                       throwable_type.str() );
+        if( !target->uncanny_dodge() ) {
```

Comment:
This compiles for me:
```suggestion
    bool uncanny = false;
    if( mattack::dodge_check( &z, target ) || ( uncanny = target->uncanny_dodge() ) ) {
        game_message_type msg_type = target->is_avatar() ? m_good : m_info;
        target->add_msg_player_or_npc( msg_type,
                                       _( "You dodge the thrown %1$s!" ),
                                       _( "<npcname> dodges the thrown %1$s!" ),
                                       throwable_type.str() );
        if( !uncanny ) {
```